bugsnag-android-core/src/main/java/com/bugsnag/android/ErrorStore.java

```diff
@@ -174,7 +175,7 @@ private void flushErrorReport(File errorFile) {
             Logger.warn("Could not send previously saved error(s)"
                 + " to Bugsnag, will try again later", exception);
         } catch (Exception exception) {
-            if (delegate != null) {
+            if (delegate != null && !(exception instanceof FileNotFoundException)) {
```

Comment:
Rather than doing `instanceof`, how about a separate `catch` block for `FileNotFoundException` that only warns? There doesn't seem any need to attempt a delete in this case either.

Reply:
Warning may also act as a useful reminder to our future selves on why we're allowing FNFE's to go unreported.